examples/ConstraintIB/eel2d/example.cpp

```diff
@@ -261,26 +308,101 @@ run_example(int argc, char* argv[])
         // Main time step loop.
         double loop_time_end = time_integrator->getEndTime();
         double dt = 0.0;
+        double current_time, new_time;
+        double box_disp = 0.0;
         while (!MathUtilities<double>::equalEps(loop_time, loop_time_end) && time_integrator->stepsRemaining())
         {
             iteration_num = time_integrator->getIntegratorStep();
             loop_time = time_integrator->getIntegratorTime();
+            current_time = loop_time;
 
             pout << "\n";
             pout << "+++++++++++++++++++++++++++++++++++++++++++++++++++\n";
             pout << "At beginning of timestep # " << iteration_num << "\n";
             pout << "Simulation time is " << loop_time << "\n";
 
             dt = time_integrator->getMaximumTimeStepSize();
-            time_integrator->advanceHierarchy(dt);
             loop_time += dt;
+            new_time = loop_time;
+
+            // Regrid the hierarchy if necessary
```

Comment:
We were finding that the drag results were smoother if the difference in momentum was computed using velocities on the same hierarchy, i.e.

1) Compute momentum at u^n
2) Advance hierarchy and regrid
3) Compute momentum at u^{n+1}
4) Compute forces from (u^{n+1}-u^n)/dt

was producing choppier force results than:

1) Regrid
2) Compute momentum at u^n
3) Advance hierarchy
4) Compute momentum at u^{n+1}
5) Compute forces from (u^{n+1}-u^n)/dt

Hence why we are regridding here: to obtain u^n on the newest hierarchy.